### PR TITLE
adding docstrings

### PR DESCRIPTION
**Pull request by esc**
This should add all remaining docstrings to the texasbbq.py script

### CODE REVIEW

File: texasbbq.py [hunk 1]
```diff
@@ -41,10 +41,12 @@
 
 
 def echo(value):
+    """ Print to stdout with prefix. """
```
Comment (stuartarchibald) [inline texasbbq.py]: Why do all these docstrings start and end with spaces?

Comment (esc) [reply texasbbq.py]: I used this style in a different project, let me use `black` here.

Comment (esc) [reply texasbbq.py]: I fixed the issue and added more missing docstrings

**Comment by esc** [comment]
All missing docstrings added and  conflicts resolved.

File: texasbbq.py [hunk 2]
```diff
@@ -218,13 +321,12 @@ def clone_url(self):
...
     @property
     def git_ref(self):
-        """ The target git ref to checkout.
+        """The target git ref to checkout.
 
         This function must work out which ref (branch or tag should be checked
         out and return that. A good start is to use
         `git_ls_remote_tags(self.clone_url)` to obtain a list of tags from the
         remote. If you want to use `master` just return that. If you specify
-        `clone_url` you should also specify this.
```
Comment (stuartarchibald) [inline texasbbq.py:227]: why delete this?

Comment (esc) [reply texasbbq.py:227]: Because the class has changed to no longer support this. You must supply a `clone_url` now. The docs are now just following suit.

Comment (stuartarchibald) [reply texasbbq.py:227]: Perhaps delete the start of the sentence too  ? :)

Comment (esc) [reply texasbbq.py:227]: Yup, just saw that and did it.

**Comment by esc** [comment]
@stuartarchibald I have addressed all the review items, is there anything else you see?

**Review by stuartarchibald — APPROVED** [review]

**Comment by stuartarchibald** [comment]
> @stuartarchibald I have addressed all the review items, is there anything else you see?

All good, thanks for the patch.

**Comment by esc** [comment]
thanks for merge

**Pull request merged**